contracts/governance/Staking/WeightedStaking.sol

```diff
@@ -589,6 +613,42 @@ contract WeightedStaking is Checkpoints {
 		emit AdminRemoved(_admin);
 	}
 
+	/**
+	 * @notice Add account to pausers ACL.
+	 * @param _pauser The address to grant pauser permissions.
+	 * */
+	function addPauser(address _pauser) public onlyOwner {
+		pausers[_pauser] = true;
+		emit PauserAddedOrRemoved(_pauser, true);
+	}
+
+	/**
+	 * @notice Add account to pausers ACL.
+	 * @param _pauser The address to grant pauser permissions.
+	 * */
+	function removePauser(address _pauser) public onlyOwner {
+		delete pausers[_pauser];
+		emit PauserAddedOrRemoved(_pauser, false);
+	}
+
+	/**
+	 * @notice Pause contract
+	 * @param _pause true when pausing, false when unpausing
+	 * */
+	function pauseUnpause(bool _pause) public onlyAuthorizedOrPauser {
+		paused = _pause;
+		emit StakingPaused(_pause);
+	}
+
+	/**
+	 * @notice Freeze contract - disable functions available even when paused
+	 * @param _freeze true when pausing, false when unpausing
```

Comment:
comment mismatch, switch pause for freeze where needed

Reply:
fixed